### PR TITLE
Version 1

### LINKED ISSUE #1: Design the quiz views
Design the views for the quiz application.

Views are:

- The start view.
- The questions view.
- The "Thank you" view.

### LINKED ISSUE #2: Implement the start button logic
The start button logic is 

```
Hide the start view and show the question view
```

### LINKED ISSUE #3: Implement the question loading
After the user pressed on the start button, load the first question from the `questions`

### Questions data structure

``` json
[
  {
    "id": 1,
    "text": "What is javascript?",
    "answers": [
      { "id": 1, "text": "It is a programming language", "is_correct": true },
      { "id": 2, "text": "It is next version of java", "is_correct": false },
      { "id": 3, "text": "It is a windows script language", "is_correct": false},
      { "id": 4, "text": "It is a name of cookies", "is_correct": false }
    ]
  },
  {...}
]

```

### LINKED ISSUE #4: Implement reveling the answer
When the user clicks on one of the answers:

1. Show whether it is correct or not to him/her. (green/red color state)
2. Show the next button.
3. If it is correct increase the score.

### LINKED ISSUE #5: Implement the next button logic
When the user clicks on the Next button, show the next question like in #3

### LINKED ISSUE #6: Implement the last question view
When the user reaches the last question and answered it. Show the total score and the exit button. The exit button will direct the user to the "Thank you" view.

### PR DESCRIPTION
**Pull request by ArwaAshawy**
closes #1
closes #2 
closes #3 
closes #4
closes #5 
closes #6

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -0,0 +1,417 @@
...
+<head>
+    <meta charset="UTF-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.0.0-beta2/dist/css/bootstrap.min.css" rel="stylesheet"
+        integrity="sha384-BmbxuPwQa2lc/FVzBcNJ7UAyJxM6wuqIj61tLrc4wSX0szH/Ev+nYRRuWlolflfl" crossorigin="anonymous">
+    <link rel="stylesheet" href="style.css">
+    <link rel="preconnect" href="https://fonts.gstatic.com">
+    <link href="https://fonts.googleapis.com/css2?family=Pacifico&display=swap" rel="stylesheet">
+    <link rel="preconnect" href="https://fonts.gstatic.com">
```
Comment (EmadMokhtar) [inline index.html:12]: Duplicate link tag

File: index.html [hunk 2]
```diff
@@ -0,0 +1,417 @@
...
+    <div id="quiz-view">
+        <div class="d-flex justify-content-center align-items-center vh-100 ">
+            <div class="container rounded px-5">
+    
+                <h1 class="logo fs-4 fw-bold my-3 neg-me">
+                    JavaScrip Quiz
+                </h1>
+                <!-- <div id="counter" class="d-flex justify-content-end hidden fw-bold">
+                    <p class="border border-1 border-dark"></p>
+                </div> -->
```
Comment (EmadMokhtar) [inline index.html]: Commented code

File: index.html [hunk 3]
```diff
@@ -0,0 +1,417 @@
...
+
+        const startBtn = document.querySelector('#start-btn');
+        const nextBtn = document.querySelector('#next-btn');
+        const exitBtn = document.querySelector('#exit-btn');
+        const question = document.querySelector('.question');
+        const answers = document.querySelector('.answers');
+        const counterDiv = document.querySelector('#counter');
+        const choices = document.querySelectorAll('.choice');
+        const quizView = document.querySelector('#quiz-view');
+        const thanksView = document.querySelector('#thanks-view')
```
Comment (EmadMokhtar) [inline index.html]: Rename vars to include the tag

Comment (EmadMokhtar) [reply index.html]: Exclude `counterDiv` `quizView` and `thanksView`

File: index.html [hunk 4]
```diff
@@ -0,0 +1,417 @@
...
+        const alertTxt =
+            "Welcome to JS quiz\n Here are the rules: \n 1 -Don't select the same answer twice.\n2- If all your answers are wrong, you will not get a feedback abput how many question(s) ypu answered.\n3- If your answer is correct, ypu will see green color\n 4- If your answer is wrong, you will see red color";
```
Comment (EmadMokhtar) [inline index.html:323]: This is a very long line. Think about a solution to make it short and readable.

File: index.html [hunk 5]
```diff
@@ -0,0 +1,417 @@
...
+            if (this === startBtn) {
+                this.classList.toggle('hidden');
+                answers.classList.toggle('hidden');
+                question.classList.toggle('hidden');
+                showQuestion();
+            } else if (this === exitBtn){
+                quizView.classList.toggle('hidden');
+                thanksView.classList.toggle('hidden');
+            }
+        }
```
Comment (EmadMokhtar) [inline index.html]: - Split the function into 2 functions (Single Responsiblity).
- Start button click ->  `startTheQuiz()`.
- End button click -> `endTheQuiz()`.

File: index.html [hunk 6]
```diff
@@ -0,0 +1,417 @@
...
+
+        function showTheResult() {
+            const p = document.createElement('p');
+            // let p = document.createElement('p');
```
Comment (EmadMokhtar) [inline index.html]: Commented code.

File: index.html [hunk 7]
```diff
@@ -0,0 +1,417 @@
...
+                question.classList.toggle('hidden');
+                showQuestion();
+            } else if (this === exitBtn){
+                quizView.classList.toggle('hidden');
+                thanksView.classList.toggle('hidden');
+            }
+        }
+
+        function showTheResult() {
+            const p = document.createElement('p');
```
Comment (EmadMokhtar) [inline index.html]:
```suggestion
            const pElement = document.createElement('p');
```

File: index.html [hunk 8]
```diff
@@ -0,0 +1,417 @@
...
+            // let p = document.createElement('p');
+            const text = document.createTextNode(`You got ${total} correct(s) from 9`);
```
Comment (EmadMokhtar) [inline index.html:346]: Choose a more descriptive name.
```suggestion
            const scoreText = document.createTextNode(`You got ${total} correct(s) from 9`);
```

File: index.html [hunk 9]
```diff
@@ -0,0 +1,417 @@
...
+            p.appendChild(text);
+            counterDiv.appendChild(p);
+        }
+        //we will remove the first element of the array after each question revealation, so in order not to modify the original array
+        let questions = quiz;
```
Comment (EmadMokhtar) [inline index.html]: Move the variable above

File: index.html [hunk 10]
```diff
@@ -0,0 +1,417 @@
...
+        let total = 0;
+        //to keep track of the question's number
+        let questionCounter = 0;
+
+        function toggleHidden(e) {
+            
+            if (this === startBtn) {
+                this.classList.toggle('hidden');
+                answers.classList.toggle('hidden');
+                question.classList.toggle('hidden');
```
Comment (EmadMokhtar) [inline index.html]: Use remove instead of a toggle. Check the below lines as well.

File: index.html [hunk 11]
```diff
@@ -0,0 +1,417 @@
...
+            const p = document.createElement('p');
+            // let p = document.createElement('p');
+            const text = document.createTextNode(`You got ${total} correct(s) from 9`);
+            p.appendChild(text);
+            counterDiv.appendChild(p);
+        }
+        //we will remove the first element of the array after each question revealation, so in order not to modify the original array
+        let questions = quiz;
+
+        function showQuestion() {
```
Comment (EmadMokhtar) [inline index.html:353]: What about adding a jsdocs to the functions? For VS code check the [link](https://code.visualstudio.com/Docs/languages/javascript#_jsdoc-support)

File: index.html [hunk 12]
```diff
@@ -0,0 +1,417 @@
...
+            counterDiv.innerHTML = `${++questionCounter}/9`;
+            question.innerHTML = questions[0].text;
+            let answers = questions[0].answers;
+            choices.forEach((choice) => {
+                backTobaseColors(choice);
+                choice.setAttribute('data-answer', `${answers[0].is_correct}`);
+                choice.setAttribute('id', `${answers[0].id}`);
+                choice.innerHTML = answers[0].text;
+                //not to repeat the sama choice twice
```
Comment (EmadMokhtar) [inline index.html]:
```suggestion
                //not to repeat the same choice twice
```

File: index.html [hunk 13]
```diff
@@ -0,0 +1,417 @@
...
+                answers.shift();
+            });
+            //not to repeat the sama question twice
```
Comment (EmadMokhtar) [inline index.html]:
```suggestion
            //not to repeat the same question twice
```

File: index.html [hunk 14]
```diff
@@ -0,0 +1,417 @@
...
+            }
+            if (this.dataset.answer == 'true') {
+                total++;
+            }
+        }
+
+        startBtn.addEventListener('click', toggleHidden);
+        startBtn.addEventListener('click', () => alert(alertTxt));
+        nextBtn.addEventListener('click', showQuestion);
+        // exitBtn.addEventListener('click', showThankYou);
```
Comment (EmadMokhtar) [inline index.html]: Commented code

File: index.html [hunk 15]
```diff
@@ -0,0 +1,417 @@
...
+        }
+
+        function wrong(elm) {
+            elm.classList.remove('bg-primary');
+            elm.classList.add('wrong');
+        }
+
+        function revealSolution() {
+            choices.forEach(choice => {
+                if (choice.dataset.answer == 'true') {
```
Comment (EmadMokhtar) [inline index.html:387]:
```suggestion
                if (choice.dataset.answer === 'true') {
```

File: index.html [hunk 16]
```diff
@@ -0,0 +1,417 @@
...
+                    wrong(choice);
+                }
+            });
+            if (questions.length !== 0) {
+                nextBtn.classList.remove('hidden');
+            } else {
+                exitBtn.classList.remove('hidden');
+                showTheResult();
+            }
+            if (this.dataset.answer == 'true') {
```
Comment (EmadMokhtar) [inline index.html:399]:
```suggestion
            if (this.dataset.answer === 'true') {
```

File: index.html [hunk 17]
```diff
@@ -0,0 +1,417 @@
...
+        }
+
+        function correct(elm) {
+            elm.classList.remove('bg-primary', 'text-white');
+            elm.classList.add('correct', 'text-dark');
+        }
+
+        function wrong(elm) {
+            elm.classList.remove('bg-primary');
+            elm.classList.add('wrong');
```
Comment (EmadMokhtar) [inline index.html]: Try to follow the BS naming convestion, for example `wrong` -> `bg-red`.

File: index.html [hunk 18]
```diff
@@ -0,0 +1,417 @@
...
+            elm.classList.add('bg-primary', 'text-white');
+            elm.classList.remove('correct', 'wrong', 'text-dark');
+        }
+
+        function correct(elm) {
+            elm.classList.remove('bg-primary', 'text-white');
+            elm.classList.add('correct', 'text-dark');
+        }
+
+        function wrong(elm) {
```
Comment (EmadMokhtar) [inline index.html]:
```suggestion
        function markWrongAnswer(elm) {
```

File: index.html [hunk 19]
```diff
@@ -0,0 +1,417 @@
...
+            questions.shift();
+            nextBtn.classList.add('hidden');
+        }
+        //to remove the modifications that were added because of revealSolution function and go back to the primary style
+        function backTobaseColors(elm) {
+            elm.classList.add('bg-primary', 'text-white');
+            elm.classList.remove('correct', 'wrong', 'text-dark');
+        }
+
+        function correct(elm) {
```
Comment (EmadMokhtar) [inline index.html]:
```suggestion
        function markCorrectAnswer(elm) {
```

**Review by EmadMokhtar — CHANGES_REQUESTED** [review]
Remove the empty js file.

File: index.html (hunk 4 above)
Comment (ArwaAshawy) [reply index.html:323]: couldn't find a solution other than this

Comment (EmadMokhtar) [reply index.html:323]: No worries this is fine.

**Review by EmadMokhtar — APPROVED** [review]
Well done

**Pull request merged**